src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -180,6 +269,21 @@ private void IssueCommand(byte[] data)
             currentSlave.Write(data);
         }
 
+        private bool SocketConnected(Socket s)
+        {
+            bool part1 = s.Poll(1000, SelectMode.SelectRead);
```

Comment:
I think it should be clearer if you rename `part1` to `dataIsAvailableForReading`